src/cljs/athens/util.cljs

```diff
@@ -2,17 +2,37 @@
   (:require
     ["/textarea" :as getCaretCoordinates]
     [clojure.string :as string]
+    [com.rpl.specter :as s]
     [goog.dom :refer [getElement setProperties]]
     [posh.reagent :refer [#_pull]]
     [tick.alpha.api :as t]
-    [tick.locale-en-us]))
+    [tick.locale-en-us])
+  (:require-macros
+    [com.rpl.specter :refer [recursive-path]]))
 
 
 (defn gen-block-uid
   []
   (subs (str (random-uuid)) 27))
 
 
+;; embed block
+
+(declare specter-recursive-path)
+
+
+(defn recursively-modify-block-for-embed
+  "Modify the block and all the block children to have same embed-id for
+   referencing the embed block rather than block in original page"
+  [block embed-id]
+  (s/transform
+    (specter-recursive-path #(contains? % :block/uid))
+    (fn [{:block/keys [uid] :as block}]
+      (assoc block :block/uid (str uid "-embed-" embed-id)
+                   :block/original-uid uid))
+    block))
```

Comment:
https://www.youtube.com/watch?v=VTCy_DkAJGk -- this will take it from 0-100 very quick

https://docs.devarkm.com/clojure/specter -- my notes and this is for final reference of how powerful it can be and how easy it is to extend it